### PR TITLE
ScrCmd_SetSlowMotion patches

### PR DESCRIPTION
**Pull request by skiff**
Made the same patches that IW4x does, which I believe came from how T6 handles slow motion? Anyways it seems to fix it up pretty well. The only issue is it is still sometimes slightly skipping frames and coming out choppy when the slowmotion is coming back to normal speed. I'm not sure if there's a way to fix that, but this is the best I could get it

### CODE REVIEW

File: src/client/module/slowmotion.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#include "utils/hook.hpp"
+#include "utils/string.hpp"
+#include "game/game.hpp"
+
+namespace slowmotion
+{
+	namespace
+	{
+		int delay = 0;
+		utils::hook::detour Com_TimeScaleMsecHook;
```
Comment (momo5502) [inline src/client/module/slowmotion.cpp:12]: pls snake case

File: src/client/module/slowmotion.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+		uint64_t Com_TimeScaleMsec(const int msec)
```
Comment (momo5502) [inline src/client/module/slowmotion.cpp:14]: pls snake case

File: src/client/module/slowmotion.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+				return Com_TimeScaleMsecHook.invoke<uint64_t>(msec);
+			}
+			else
+			{
+				delay -= msec;
+				return 0;
+			}
+		}
+
+		void ScrCmd_SetSlowMotion()
```
Comment (momo5502) [inline src/client/module/slowmotion.cpp:27]: pls snake case

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/module/patches.cpp [hunk 4]
```diff
@@ -318,6 +318,25 @@ namespace patches
...
+						{
+							auto mod_plugin = utils::nt::module::load(file);
+							if (mod_plugin.is_valid())
+							{
+								printf("Plugin \"%s\" has been loaded\n", mod_plugin.get_name().data());
+							}
+						}
+					}
+				}
+			}, scheduler::pipeline::main);
```
Comment (momo5502) [inline src/client/module/patches.cpp:339]: Not sure if you were supposed to commit that ;)

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/module/patches.cpp (hunk 4 above)
Comment (skiff) [reply src/client/module/patches.cpp:339]: oops lmfao I'll remove it

**Review by momo5502 — APPROVED** [review]

**Pull request merged**